web_src/js/features/imagediff.js

```diff
@@ -133,24 +133,17 @@ export function initImageDiff() {
         $container.find('.bounds-info-before .bounds-info-height').text(`${sizes.image2[0].naturalHeight}px`).addClass(heightChanged ? 'red' : '');
       }
 
-      sizes.image1.css({
-        width: sizes.size1.width * factor,
-        height: sizes.size1.height * factor
-      });
-      sizes.image1.parent().css({
-        margin: `10px auto`,
-        width: sizes.size1.width * factor + 2,
-        height: sizes.size1.height * factor + 2
-      });
-      sizes.image2.css({
-        width: sizes.size2.width * factor,
-        height: sizes.size2.height * factor
-      });
-      sizes.image2.parent().css({
-        margin: `10px auto`,
-        width: sizes.size2.width * factor + 2,
-        height: sizes.size2.height * factor + 2
-      });
+      sizes.image1[0].style.width = `${sizes.size1.width * factor}px`;
+      sizes.image1[0].style.height = `${sizes.size1.height * factor}px`;
+      sizes.image1.parent()[0].style.margin = '10px auto';
+      sizes.image1.parent()[0].style.width = `${sizes.size1.width * factor + 2}px`;
+      sizes.image1.parent()[0].style.height = `${sizes.size1.height * factor + 2}px`;
+
+      sizes.image2[0].style.width = `${sizes.size2.width * factor}px`;
+      sizes.image2[0].style.height = `${sizes.size2.height * factor}px`;
+      sizes.image2.parent()[0].style.margin = '10px auto';
+      sizes.image2.parent()[0].style.width = `${sizes.size2.width * factor + 2}px`;
+      sizes.image2.parent()[0].style.height = `${sizes.size2.height * factor + 2}px`;
```

Comment:
It doesn't look good to me.

1. `image1` and `image2` might not exist.
2. Since this is a rewriting and it needs a full test, I think it's better to rewrite it with a formal coding style, but not keeping the old style (the old style is good in jQuery, but fragile by DOM JS)

Reply:
Pretty sure hese images are guaranteed to exist in all 3 tabs of the image diff. I don't really want to touch this messy code any more than necessary, but I guess some `if could` be added around.

Reply:
> > image1 and image2 might not exist.
> 
> Pretty sure these images are guaranteed to exist in all 3 tabs of the image diff.

I do not think so. See L127 and L131 .

> I don't really want to touch this messy code any more than necessary, but I guess some .? could be added if that makes you feel better.

`?` would make the code more messy. I would still suggest to rewrite the code properly, since it needs to be fully tested.

Reply:
I can wrap in `if` as well to replicate jQuery's behaviour of ignoring non-existant elements. Not in a mood of rewriting  this completely right now.

Reply:
That's why I can't totally agree with your "jQuery removal plan". If people only rewrite code blindly, it just introduces bugs and messy code.

Reply:
> `.?` accurately pictures jquery's behaviour of empty collections. It's not "more messy". It's just a correct replacement.

I could compromise for this case, while actually a lot of `sizes.image1[0]?.xxx` and `sizes.image2.parent()[0]?.xxx` don't really look good to me.

I think the proper rewriting could be like this:

```
const sizesImage1 = sizes.image[0];
if (sizesImage1) {
    sizesImage1.xxx = 
    sizesImage1.parentNode.xxx = 
}
```

Reply:
Cleaned it up in https://github.com/go-gitea/gitea/pull/29852/commits/8b6199ccade147e68919ee1e65df6213efa9a6a8.